Assets/Global/Scripts/Player/Tails/Tail.cs

```diff
@@ -1,44 +1,74 @@
 using System;
+using System.Collections.Generic;
+using System.Linq;
 using UnityEngine;
 
+
 public class Tail : MonoBehaviour
 {
     public Player player;
     public BaseTail currentTail;
-    public Animator animator;
+    public List<BaseTail> tails;
+
+    public List<Attack> attacks;
+
+    public Animator WaffleAnimator;
 
     [HideInInspector]
     public int attackIndex;
 
     [HideInInspector]
-    public int tailDoDamage;
+    public float tailDoDamage;
 
     [HideInInspector]
-    public float activeAttackCooldown;
+    public float activeResetComboTime;
+
+    public float comboResetTime = 2f;
+
+    public float activeCooldownTime;
+    public float cooldownTime = 0.0f;
 
     [HideInInspector]
     public bool tailCanDoDamage = false;
-    public float attackResettingTime = 2f;
+    public GameObject slamObject;
+    public bool flipDoDamage = false;
+
+    public float increaseTailSpeed = 1.0f;
 
     public void Update()
     {
-        activeAttackCooldown =
+        activeResetComboTime =
             player.currentState.GetType().Name != "AttackingState"
-                ? activeAttackCooldown + Time.deltaTime
+                ? activeResetComboTime + Time.deltaTime
                 : 0.0f;
-        if (activeAttackCooldown >= attackResettingTime)
+        if (activeResetComboTime >= comboResetTime)
         {
             attackIndex = 0;
-            activeAttackCooldown = 0.0f;
+            activeResetComboTime = 0.0f;
         }
+
+        activeCooldownTime =
+            player.currentState.GetType().Name != "AttackingState"
+                ? activeCooldownTime + Time.deltaTime
+                : activeCooldownTime;
     }
 
-    public void ChangeTail(BaseTail newtail, Animator animator)
+    public void WaffleQuake()
```

Comment:
This is not how we want it in the futute.
We are going to have a lot more upgrades in the future. Not only for the tail, but also for other things>

Things like the scale should be a stat so we dont change it like this. And upgrades should be its own system of some sorts so we can easaly strip away a stake without risking to fuck up the current internal state. (state as in, all values of the object, not the actual state of  the state machine)

But this is something for in the future. For now hardcoded like this is fine